public/app/plugins/panel/logs/LogsPanel.tsx

```diff
@@ -465,62 +471,112 @@ export const LogsPanel = ({
+              rows={logRows}
+              scrollElement={scrollElement}
+              sortOrder={sortOrder}
+            >
+              <LogRows
+                scrollElement={scrollElement}
+                scrollIntoView={scrollIntoView}
+                permalinkedRowId={getLogsPanelState()?.logs?.id ?? undefined}
+                onPermalinkClick={showPermaLink() ? onPermalinkClick : undefined}
+                logRows={logRows}
+                showContextToggle={showContextToggle}
+                deduplicatedRows={deduplicatedRows}
+                dedupStrategy={dedupStrategy}
+                showLabels={showLabels}
+                showTime={showTime}
+                wrapLogMessage={wrapLogMessage}
+                prettifyLogMessage={prettifyLogMessage}
+                timeZone={timeZone}
+                getFieldLinks={getFieldLinks}
+                logsSortOrder={sortOrder}
+                enableLogDetails={enableLogDetails}
+                previewLimit={isAscending ? logRows.length : undefined}
+                onLogRowHover={onLogRowHover}
+                app={isCoreApp(app) ? app : CoreApp.Dashboard}
+                onOpenContext={onOpenContext}
+                onClickFilterLabel={
+                  isOnClickFilterLabel(onClickFilterLabel) ? onClickFilterLabel : defaultOnClickFilterLabel
+                }
+                onClickFilterOutLabel={
+                  isOnClickFilterOutLabel(onClickFilterOutLabel) ? onClickFilterOutLabel : defaultOnClickFilterOutLabel
+                }
+                onClickFilterString={isOnClickFilterString(onClickFilterString) ? onClickFilterString : undefined}
+                onClickFilterOutString={
+                  isOnClickFilterOutString(onClickFilterOutString) ? onClickFilterOutString : undefined
+                }
+                isFilterLabelActive={isIsFilterLabelActive(isFilterLabelActive) ? isFilterLabelActive : undefined}
+                displayedFields={displayedFields}
+                onClickShowField={displayedFields !== undefined ? onClickShowField : undefined}
+                onClickHideField={displayedFields !== undefined ? onClickHideField : undefined}
+                logRowMenuIconsBefore={isReactNodeArray(logRowMenuIconsBefore) ? logRowMenuIconsBefore : undefined}
+                logRowMenuIconsAfter={isReactNodeArray(logRowMenuIconsAfter) ? logRowMenuIconsAfter : undefined}
+                // Ascending order causes scroll to stick to the bottom, so previewing is futile
+                renderPreview={isAscending ? false : true}
+              />
+            </InfiniteScroll>
+            {showCommonLabels && isAscending && renderCommonLabels()}
+          </div>
+        </ScrollContainer>
+      ) : (
+        <div onMouseLeave={onLogContainerMouseLeave} className={style.controlledLogsContainer}>
           {showCommonLabels && !isAscending && renderCommonLabels()}
-          <InfiniteScroll
+          <ControlledLogRows
```

Comment:
ControlledLogRows owns the scrolling div, so we're not wrapping this in ScrollContainer for plugins with enabled controls.